akka-actor/src/main/scala/akka/util/ManifestInfo.scala

```diff
@@ -80,11 +80,18 @@ object ManifestInfo extends ExtensionId[ManifestInfo] with ExtensionIdProvider {
     if (values.size > 1) {
       val highestVersion = values.max
       val toBeUpdated = filteredVersions.collect { case (k, v) if v != highestVersion => s"$k" }.mkString(", ")
+      val groupedByVersion = filteredVersions.toSeq
+        .groupBy { case (_, v) => v.version }
+        .view
+        .map { case (k, v) => k -> v.map(_._1).mkString("[", ", ", "]") }
+        .map { case (k, v) => s"($k, $v)" }
+        .mkString(", ")
       Some(
         s"You are using version $highestVersion of $productName, but it appears " +
         s"you (perhaps indirectly) also depend on older versions of related artifacts. " +
         s"You can solve this by adding an explicit dependency on version $highestVersion " +
         s"of the [$toBeUpdated] artifacts to your project. " +
+        s"Here's a complete collection of detected artifacts: ${groupedByVersion}. " +
         "See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed")
```

Comment:
This informs of all artifacts considered in the decision process but doesn't improve the message re what alternatives the user has and what is the preferred alternative in most cases (keep the highest akka version).